src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi

```diff
@@ -36,23 +36,25 @@ cdef class CallbackWrapper:
         self.context.functor.functor_run = self.functor_run
         self.context.waiter = <cpython.PyObject*>future
         self.context.failure_handler = <cpython.PyObject*>failure_handler
+        self.context.callback_wrapper = <cpython.PyObject*>self
         # NOTE(lidiz) Not using a list here, because this class is critical in
         # data path. We should make it as efficient as possible.
         self._reference_of_future = future
         self._reference_of_failure_handler = failure_handler
+        cpython.Py_INCREF(self)
```

Comment:
As documentation I would put something here explaining the reason of holding an artificial reference more for this object.